presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestConnectorOptimization.java

```diff
@@ -284,7 +284,7 @@ private static void assertPlanMatch(PlanNode actual, PlanMatchPattern expected,
 
     private static PlanNode optimize(PlanNode plan, Map<ConnectorId, Set<ConnectorPlanOptimizer>> optimizers)
     {
-        ApplyConnectorOptimization optimizer = new ApplyConnectorOptimization(optimizers);
+        ApplyConnectorOptimization optimizer = new ApplyConnectorOptimization(() -> optimizers);
```

Comment:
add a comment about why use `Supplier` here

Reply:
I assume you mean [here](https://github.com/prestodb/presto/pull/12994/files#diff-cf435c810225e315249d564757f3d5d3R73) because the one you commented is a test?